base/array.jl

```diff
@@ -36,6 +36,8 @@ end
 ## copy ##
 
 function unsafe_copy!{T}(dest::Ptr{T}, src::Ptr{T}, n)
+    # Do not use this to copy data between pointer arrays.
+    # It can't be made safe no matter how carefully you checked.
```

Comment:
For someone like me not in this business, this comment is a bit frightening. Basically when I use this function, `dest` and `src` point to arrays, in the C meaning, so I interpret this comment as "never use this function", because my goal is precisely to copy data between "pointer arrays". So it would be useful to clarify your warning, and to put it in the docstring, which is more visible.

Reply:
Ah sorry, you meant arrays of pointer! still, would be worth it to move the warning in the docstring.